auth/handlers/handle_login_test.go

```diff
@@ -0,0 +1,184 @@
+package handlers
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/cortezaproject/corteza-server/auth/request"
+	"github.com/cortezaproject/corteza-server/auth/settings"
+	"github.com/cortezaproject/corteza-server/system/service"
+	"github.com/cortezaproject/corteza-server/system/types"
+	"github.com/quasoft/memstore"
+	"github.com/stretchr/testify/require"
+)
+
+type (
+	expectPayload struct {
+		kv    map[string]string
+		alert []request.Alert
+	}
+)
+
+func Test_loginForm_setValues(t *testing.T) {
+	var (
+		ctx      = context.Background()
+		memStore = memstore.NewMemStore()
+		user     = makeMockUser(ctx)
+
+		req = &http.Request{}
+
+		authService  *mockAuthService
+		authHandlers *AuthHandlers
+		authReq      *request.AuthReq
+
+		rq = require.New(t)
+	)
+
+	service.CurrentSettings = &types.AppSettings{}
+	service.CurrentSettings.Auth.Internal.Enabled = true
+
+	authSettings := &settings.Settings{}
+
+	authService = prepareClientAuthService(ctx, user, memStore)
+	authReq = prepareClientAuthReq(ctx, req, user, memStore)
+	authHandlers = prepareClientAuthHandlers(ctx, authService, authSettings)
+
+	payload := map[string]string{"key": "value"}
+	authReq.SetKV(payload)
+
+	authHandlers.Settings = &settings.Settings{
+		EmailConfirmationRequired: true,
+	}
+
+	err := authHandlers.loginForm(authReq)
+
+	rq.NoError(err)
+	rq.Equal(TmplLogin, authReq.Template)
+	rq.Equal(payload, authReq.Data["form"])
+}
+
+func Test_loginProc(t *testing.T) {
+	var (
+		ctx      = context.Background()
+		memStore = memstore.NewMemStore()
+		user     = makeMockUser(ctx)
+
+		req = &http.Request{}
+
+		authService  authService
+		authHandlers *AuthHandlers
+		authReq      *request.AuthReq
+
+		authSettings = &settings.Settings{}
+
+		rq = require.New(t)
```

Comment:
should probably be moved inside sub-test fn. I had problems when I used one initialized on root.